Core/GDCore/Events/CodeGeneration/EventsCodeGenerationContext.h

```diff
@@ -269,10 +308,13 @@ class GD_CORE_API EventsCodeGenerationContext {
                               ///< inheriting from context with depth n.
   unsigned int
       customConditionDepth;  ///< The depth of the conditions being generated.
+  unsigned int asyncDepth;
   unsigned int* maxDepthLevel;  ///< A pointer to a unsigned int updated with
                                 ///< the maximum depth reached.
   const EventsCodeGenerationContext*
       parent;  ///< The parent of the current context. Can be NULL.
+  EventsCodeGenerationContext*
+      nearestAsyncParent = nullptr;  ///< The nearest context that 

```

Comment:
I have initialized the variable with this, but the other variables are initialized in the constructor, is there a reason for that? Initializing in the constructor looks error-prone to me since one could forget to initialize a value when making a new constructor, where using default values like this allow to always have a valid default value.

Reply:
Reason is mostly historical/old habit (before C++11/14 you could not initialise members directly in the class declaration, or support in compilers was not even).
Now it's fine to do that :)

Reply:
Ah, I see, thanks for the explanation!